src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke.main;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+import duke.exception.DeadlineIncompleteException;
+import duke.exception.DeleteIncompleteException;
+import duke.exception.DeleteOutOfListException;
+import duke.exception.DoneIncompleteException;
+import duke.exception.DoneOutOfListException;
+import duke.exception.DukeException;
+import duke.exception.EventIncompleteException;
+import duke.exception.FindIncompleteException;
+import duke.exception.NoInputException;
+import duke.exception.TodoIncompleteException;
+import duke.exception.UnknownInputException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Parser is used to make sense the input from the user and convert it to executable command.
+ */
+public class Parser {
+
+    public static String[] commands = {"list", "done", "deadline", "event", "todo", "delete", "bye", "find"};
+    public static ArrayList<String> VALID_COMMAND = new ArrayList<>(Arrays.asList(commands));
+
+    /**
+     * Checks the correctness of the input by the user.
+     *
+     * @param input The input from the user.
+     * @throws DukeException If user's input is invalid.
+     */
+    public static void checkInput(String ... input) throws DukeException {
+        String command = input[0];
+        System.out.println(command);
+        if (command.equals("")) {
+            throw new NoInputException();
+        } else if (!VALID_COMMAND.contains(input[0])) {
+            throw new UnknownInputException();
+        } else if (input.length == 1) {
+            switch (command) {
+                case "done":
+                    throw new DoneIncompleteException();
+                case "deadline":
+                    throw new DeadlineIncompleteException();
+                case "event":
+                    throw new EventIncompleteException();
+                case "todo":
+                    throw new TodoIncompleteException();
+                case "delete":
+                    throw new DeleteIncompleteException();
+                case "find":
+                    throw new FindIncompleteException();
```

Comment:
The `case`s have a wrong indentation level, as per SEEDU's java coding standard. Unfortunately this one is one that IntelliJ won't catch :(